lib/checks/hasUnsupportedNodes.js

```diff
@@ -1,30 +1,29 @@
-const Node = require('../node');
+const {query} = require('math-nodes');
+
 const resolvesToConstant = require('./resolvesToConstant');
 
 function hasUnsupportedNodes(node) {
-  if (Node.Type.isParenthesis(node)) {
-    return hasUnsupportedNodes(node.content);
-  }
-  else if (Node.Type.isUnaryMinus(node)) {
-    return hasUnsupportedNodes(node.args[0]);
-  }
-  else if (Node.Type.isOperator(node)) {
-    return node.args.some(hasUnsupportedNodes);
-  }
-  else if (Node.Type.isSymbol(node) || Node.Type.isConstant(node)) {
-    return false;
-  }
-  else if (Node.Type.isFunction(node, 'abs')) {
-    if (node.args.length !== 1) {
-      return true;
+  if (query.isApply(node)) {
+    if (query.isAbs(node)) {
+      if (node.args.length !== 1) {
+        return true;
+      }
+      if (hasUnsupportedNodes(node.args[0])) {
+        return true;
+      }
+      if (!resolvesToConstant(node.args[0])) {
```

Comment:
TBH, I don't know why this is the case.  I was copying what was on line 24.  Personally I'm okay with having non-constant nodes inside of an absolute.  The only thing is that if they are there we can't evaluate it, but it seems like we'd probably want a separate function called `canEvaluate` to check for that.  I'll get rid of this line if you're okay with that.